cli/replicaset/cmd/roles.go

```
@@ -0,0 +1,121 @@
+	RunningCtx running.RunningCtx
+	// Force true if unfound instances can be skipped.
+	Force bool
+	// Timeout describes a timeout in seconds.
+	// We keep int as it can be passed to the target instance.
+	Timeout int
+}
+
+// RolesAdd adds role with provided path target to config.
+func RolesAdd(ctx RolesAddCtx) error {
+	var (
+		orchestratorType replicaset.Orchestrator
+		err              error
+	)
+	orchestratorType, err = getInstanceOrchestrator(ctx.Orchestrator, ctx.Conn)
+	if err != nil {
+		return err
+	}
+
+	if orchestratorType == replicaset.OrchestratorCartridge {
+		if ctx.ReplicasetName == "" {
+			return fmt.Errorf("in cartridge need to specify replica set name via --replicaset flag")
+		}
+	}
+
+	var orchestrator replicasetOrchestrator
+	if ctx.IsApplication {
+		if orchestrator, err = makeApplicationOrchestrator(
+			orchestratorType, ctx.RunningCtx, ctx.Collectors, ctx.Publishers); err != nil {
+			return err
+		}
+	} else {
+		if orchestrator, err = makeInstanceOrchestrator(orchestratorType, ctx.Conn); err != nil {
+			return err
+		}
+	}
+
+	log.Info("Discovery application...")
+	fmt.Println()
+
+	// Get and print status.
+	replicasets, err := orchestrator.Discovery(replicaset.SkipCache)
+	if err != nil {
+		return err
+	}
+	statusReplicasets(replicasets)
+	fmt.Println()
+
+	if ctx.IsGlobal {
+		if orchestratorType == replicaset.OrchestratorCartridge {
+			log.Warn("Ignoring --global (-G) flag due to cluster with cartridge" +
+				" orchestrator can't add role globally")
+		} else {
+			log.Infof("Add role %s to global scope", ctx.RoleName)
+		}
+	}
+	if ctx.GroupName != "" && orchestratorType != replicaset.OrchestratorCartridge {
+		log.Infof("Add role %s to group: %s", ctx.RoleName, ctx.GroupName)
+	}
+	if ctx.InstName != "" {
+		if orchestratorType == replicaset.OrchestratorCartridge {
+			log.Warn("Ignoring --instance (-i) flag due to cluster with cartridge orchestrator" +
+				" can't add role into instance scope")
```

I think we should return an error in such cases.

---

Now the program returns error in such cases.